RealifeTech-CoreSDK/APILayer/Util/String+RequestIdentifier.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  String+RequestIdentifier.swift
+//  RealifeTech-CoreSDK
+//
+//  Created by Mickey Lee on 05/03/2021.
+//
+
+import Foundation
+import CryptoKit
+
+extension String {
+
+    var sha256: String {
+        let digest = SHA256.hash(data: self.data(using: .utf8) ?? Data())
```

Comment:
Can the return statement be removed here as well as the reference to self by doing following:
```
var sha256: String {
    SHA256.hash(data: data(using: .utf8) ?? Data()).hexString
}
```

Reply:
If `data(using: .utf8)` fails, what does hashing `Data()` produce? Is it an empty string?

Reply:
I think it won't return an empty string if hashing Data(). Probably it'd be better to use a guard statement to ensure it returns an empty string like below. The data is non-optional in `SHA.hash(data: DataProtocol)` so it's unlikely to just use `SHA256.hash(data: data(using: .utf8) ?? Data()).hexString`
```
guard let data = self.data(using: .utf8) else { return "" }
let digest = SHA256.hash(data: data)
return digest.hexString
```

Reply:
The code I wrote was doing the same as your original code. I was just curious what it produced it `data(using: .utfi)` failed. Returning an empty string is probably better, as encrypting empty data doesn't seem useful.

Could use map:
```
data(using: .utf8).map { SHA.hash(data: $0).hexString } ?? ""
```